net/csshnpd/Makefile

```diff
@@ -0,0 +1,74 @@
+#
+# Copyright (C) 2025 The Atsign Foundation
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=csshnpd
+PKG_VERSION:=1.0.8
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-c$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/atsign-foundation/noports/releases/download/c$(PKG_VERSION)
+PKG_HASH:=f579601e47c766ea1531b5d98702c6d7111c700575ae11360b2672c9f9b39d91
+
+PKG_MAINTAINER:=Chris Swan <chris@atsign.com>
+PKG_LICENSE:=BSD-3-Clause
+PKG_LICENSE_FILES:=LICENCE
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-c$(PKG_VERSION)
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+CMAKE_OPTIONS += \
```

Comment:
Most of the cmake flags are set automatically during `include $(INCLUDE_DIR)/cmake.mk`  [include](https://github.com/openwrt/openwrt/blob/7193539c9811f5a3222fcff553e1193c2e451969/include/cmake.mk#L96). So not all of them you have set are needed and must be set.

Reply:
There weren't many dupes, but I've taken out what I could find in https://github.com/openwrt/packages/pull/26518/commits/2b080148ae6833ed7d8ae8c5dd14b05b272ecf12

The reason we've got a lot of flags there is we're using CPack to create a source tarball that carries the various dependencies, so we're not trying to pull stuff in from git at build time. Many of the flags are for those dependencies, and we've had to copy over all the flags from the CPack presets as we can't directly invoke a preset from the Makefile (which would make the CMake options very short).

Reply:
If I understand the structure of your package correctly and I can only speak for myself now, but loading any dependencies implicitly when building in a cross build system is not so good idea.

This is probably about ?
```
-DNOPORTS_MBEDTLS_PATH="deps/mbedtls-src" \
-DNOPORTS_CJSON_PATH="deps/cjson-src" \
-DNOPORTS_ATSDK_PATH="deps/atsdk-src"
```
From my point of view, we should also package this separately.
Of course, I could also be wrong.
Perhaps you can explain it to me in more detail.

Reply:
I would prefer dynamic linking.
It also has security aspects.
If an error is detected in the dynamically linked mbedtls, then only this needs to be replaced.

Reply:
The dynamically linked version is in https://github.com/openwrt/packages/pull/26518/commits/d326b2201cd3acaab9d11301349a863fda221e1e

I'd appreciate your thoughts on why the mips_24kc test is failing? Looks like an underlying problem with cJSON:

```log
Testing package csshnpd in version 1.0.10 from net/csshnpd
No pre-test.sh script available
(1/2) Installing cJSON (1.7.18-r1)
ERROR: cJSON-1.7.18-r1: ADB block error
(2/2) Installing csshnpd (1.0.10-r1)
Bus error (core dumped)
Error: Process completed with exit code 138.
```

Reply:
This is an error from [APK](https://gitlab.alpinelinux.org/alpine/apk-tools/-/blob/master/src/print.c#L66).
I assume there is something wrong with the `APK` installation package of `cJSON` with this architectures.